File: mq/rabbitmq/php/src/broadcast/publisher.php
<?php
/**
 * rabbitmq hello-world
 */
require_once __DIR__ . '/../bootstrap.php';


$connection = new \PhpAmqpLib\Connection\AMQPStreamConnection(RABBITMQ_MASTER, 5672, 'guest', 'guest');
$channel = $connection->channel();

$channel->exchange_declare('logs_broadcast', 'fanout', false, true, false);

$name = bin2hex(random_bytes(4));
for ($index = 0; $index < 1 << 32; ++$index) {
    $message = json_encode(['producer' => $name, 'index' => $index, 'from' => 'php', 'body' => bin2hex(random_bytes(16))]);
    $channel->basic_publish(new \PhpAmqpLib\Message\AMQPMessage($message), 'logs_broadcast');
    echo sprintf("publish <%s@%s> from php\n", $name, $index);
    sleep(random_int(1, 5));
}

File: mq/rabbitmq/py/broadcast/publisher.py
#!/usr/bin/env python3
#
# Copyright (C) 2019 jayson
#
import json
import logging
import utils.logging
from utils.connection import get_exchange_connection
from utils.random import random_string, random_sleep
from config import RABBITMQ_MASTER


if __name__ == '__main__':
    utils.logging.init_logging(logging.DEBUG)
    with get_exchange_connection(RABBITMQ_MASTER, 'logs_broadcast', 'fanout') as channel:
        name = random_string(8)
        for index in range(2 ** 32):
            message = json.dumps({'producer': name, 'index': index, 'from': 'python', 'body': random_string(32)})
            channel.basic_publish(exchange='logs_broadcast', routing_key='', body=message.encode('utf-8'))
            utils.logging.info("send message<{}@{}> to logs_broadcast from python".format(name, index))
            random_sleep(5)
